form-libraries/formik.mdx

```diff
@@ -0,0 +1,84 @@
+---
+name: Formik
+menu: 4. Form Libraries
+route: /formik
+---
+
+# Integrating Fonk with React Formik
+
+[Formik](https://github.com/jaredpalmer/formik) is a great library to manage forms state.
+
+Adding **Fonk** to **Formik** allows you to seamlessly add validation capabilities to **Formik**.
+
+In order to use **Fonk** with **Formik** you will have to install **fonk-formik** adaptor:
+
+```bash
+npm install @lemoncode/fonk @lemoncode/fonk-formik--save
+```
+
+The main change to apply when using a **fonk-formik** comes when you want to instantiate
+**Fonk** engine with your form validatin schema, instead of calling _createFormValidation_
+, just replace that entry with _createFonkValidation_
+
+```diff
+- import { createFormValidation, Validators } from '@lemoncode/fonk';
++ import { Validators } from '@lemoncode/fonk';
++ import { createFonkValidation } from '@lemoncode/fonk-formik';
+import { isNumber } from '@lemoncode/fonk-is-number-validator';
+import { minNumberValidator } from './custom-validators';
+
+const validationSchema = {
+  field: {
+    email: [Validators.required.validator],
+    password: [Validators.required.validator]
+  }
+};
+
+- export const formValidation = createFormValidation(validationSchema);
++ export const formValidation = createFonkValidation(validationSchema);
+```
+
+Now you can hook to Formik form validation (example):
+
+```diff
+    <Formik
+      initialValues={{ email: "", password: "" }}
++       validate={formValidation.validateForm}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          alert(JSON.stringify(values, null, 2));
+          setSubmitting(false);
+        }, 400);
+      }}
+    >
+```
+
+And if you want to hook to Formik field validations (example):
+
+```diff
+    <Field name="email" validate={validateField("email")} />
++    {errors &&
++      errors.email &&
++       touched.email &&
++       <div>{errors.email}</div>}
+```
+
+# Examples:
+
+[Basic example](https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/formik/js/basic)
+
+[Using formik Field](https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/react-final-form/js/formik-component)
```

Comment:
Typo `formik-components`. The folder name is with `s`

Reply:
Typo in url too.

Reply:
thanks